cli/cmd/sync_v2.go

```diff
@@ -153,6 +153,17 @@ func syncConnectionV2(ctx context.Context, sourceClient *managedplugin.Client, d
 		}
 	}
 	for i := range destinationsClients {
+		if destinationSpecs[i].WriteMode == specs.WriteModeOverwriteDeleteStale {
```

Comment:
same here, alloc `deleteStaleMsg` outside of loop, reuse)

Reply:
Personally I find it easier to follow when the allocation is close to where it's used, and I don't think we expect `destinationsClients` to be very large